api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -80,6 +81,28 @@ public interface DeleteOrphanFiles extends Action<DeleteOrphanFiles, DeleteOrpha
    */
   DeleteOrphanFiles executeDeleteWith(ExecutorService executorService);
 
+
+  /**
+   * Pass a mode for handling the files that cannot be determined if they are orphan
+   * @param mode mode for handling files that cannot be determined if they are orphan
+   * @return this for method chaining
+   */
+  DeleteOrphanFiles prefixMismatchMode(String mode);
+
+  /**
+   * Pass a list of schemes to be considered equivalent when finding orphan files
+   * @param equivalentSchemes list of equivalent schemes
+   * @return this for method chaining
+   */
+  DeleteOrphanFiles equivalentSchemes(List<String> equivalentSchemes);
+
+  /**
+   * Pass a list of authorities to be considered equivalent when finding orphan files
```

Comment:
Question: what if I have the following use case where `bucket1` and `bucket2` are indeed different?

```
s3://bucket1/path/to/file.parquet - actual
s3://bucket2/path/to/file.parquet - referenced by metadata
```

The first file should be deleted. How can we support this case with this API? The only way I can think of right now is to add another method of unequal authorities. That seems to complicate the action quite a bit.

Any better ideas?

Reply:
I spent more time thinking and couldn't come up with anything better than `unequalAuthorities`.
Do you have any ideas, @karuppayya @flyrain @RussellSpitzer @szehon-ho @rdblue?

Reply:
So this is the issue in which you have the same path in multiple authorities but only one authority is correct? BUT you are also passing in both authorities to Delete-orphans to search for file listing, I didn't think we could do this? Seems like a rare use case.

Wouldn't a user either being using s3://bucket1 or s3://bucket2 as the root search path for removing files? If they are passing in bucket1 they probably shouldn't pass in bucket2 as an acceptable equivalent.

Reply:
@RussellSpitzer, let me give an example. Suppose you have two absolutely different buckets (`b1` and `b2`). You write new data into `b1` but periodically clone files from `b1` into `b2`. The table contains some files in `b1` and some files in `b2`. Now image one clone job failed but wrote a file in `b2`.

```
s3://b2/path/to/file.parquet
```

Whenever you decide to clean `b2` against the table metadata, this file should be considered orphan. The metadata references the original file in `b1` as the clone job failed.

```
s3://b1/path/to/file.parquet
```

Under the current design, there is no way to clean that file. I don't have a real use case for this so the question is whether we should worry about it.

Reply:
Wouldn't you not pass the two buckets here since they aren't equivalent? If they are different storage locations, then you would never tell this that they are equivalent.

Reply:
@rdblue, correct, we will not pass these two to `equivalentAuthorities`. That means there will be an exception saying we detected a case when the file paths are identical except the authority. How do we tell the action to delete the file in this case? We may assume if the user provided a map of equivalent authorities, then all other authorities can be treated unequal.

Reply:
> We may assume if the user provided a map of equivalent authorities, then all other authorities can be treated unequal.

This seems to be the best idea I have so far.